### PR TITLE
Fix package manager detection

### PR DESCRIPTION
**Pull request by sparknsh**

### CODE REVIEW

**Comment by sparknsh** [comment]
@murrant I have adjusted the script to still use the env variables.

**Comment by sparknsh** [comment]
@murrant or @laf Can you merge this in?

File: snmp/osupdate [hunk 1]
```diff
@@ -28,39 +28,39 @@ CMD_PACMAN='-Sup'
...
+elif [ -f '/etc/arch-release' ]; then
     # Arch
     UPDATES=`$BIN_PACMAN $CMD_PACMAN | $BIN_WC $CMD_WC`
     if [ $UPDATES -ge 1 ]; then
         echo $(($UPDATES-1));
     else
         echo "0";
     fi
-elif [ -f "$BIN_YUM" ]; then
+elif [ -f '/etc/redhat-release' ]; then
```
Comment (murrant) [inline snmp/osupdate]: This doesn't cover centos.

Comment (sparknsh) [reply snmp/osupdate]: It sure does. 

`cat /etc/redhat-release
CentOS Linux release 7.5.1804 (Core)`

Comment (murrant) [reply snmp/osupdate]: Let me be more generic.  This doesn't include some random distro that uses yum and doesn't have /etc/redhat-release.  Replace yum and /etc/redhat-release with any combo.

Comment (sparknsh) [reply snmp/osupdate]: Made the changes

**Comment by murrant** [comment]
Getting warmer here, but I think it is more portable to test the return code of command -v instead of the output.

**Comment by murrant** [comment]
Thanks for working on this @sparknsh 

After a bit of testing, I _think_ this is what you want:
```
if command -v ls &>/dev/null ; then echo 'yes'; fi
```

**Comment by sparknsh** [comment]
@murrant can you merge this in?

**Comment by murrant** [comment]
Sure

**Pull request merged**